### DISCUSSION TITLE
Using hourly accumulated precipitation observation data

### DISCUSSION THREAD

**Discussion by maplegilbs** [weather-gov/api#463] [Q&A]
Before I begin I should state  just how much I appreciate this amazing service being offered by NOAA to the public at no charge. 

For reference and example of the api endpoint I am accessing for the is the Burlington VT station is 
https://api.weather.gov/stations/KBTV/observations

Here is my issue.  I am attempting to write a script that can determine precip totals at this station for any given time interval.  I.e for the past hour, two hours, three hours etc.  So if I make that api call at say, 11:20 am, and the most recent observational data returned is from 11:15, the result of my script using the returned data would be a 1hr total of rain between 10:15 and 11:15, a 2hr hour total from 9:15 am to 11:15 am, a 3hr totalling from 8:15 to 11:15 etc.

I am finding this difficult to execute and here is why:

The JSON I am seeing is has the key=> value pair precipitationLastHour: {precip}.  Which I initially interpreted to mean a total precip recorded in the last hour.  Upon further inspection I now interpret this to mean total precip recorded since the last 'breakpoint' where the breakpoints are a) unique to each station b) do not fall on the hour or other obvious interval.  For example as far as I can tell the breakpoints for the Burlington station are at 54 minutes past the hour.  Therefore the precipitationLastHour value I would expect to find from the api call descripted previously (most recent observation data returned being from 11:15) would only be the total accumulated from between 10:54 and 11:15.  

This would not be an issue if I knew where these breakpoints were for each station - as a matter of fact it would make my life much easier.  Unfortunately without knowing, it is very hard to tell my script which "precipitationLastHour" values to total in order to arrive my desired reporting intervals.

I would greatly appreciate any insight into how to resolve this - specifically if there is any information contained in the JSON that would indicate when the accumulator for the precipitationLastHour is reset / when the breakpoints are for each individual station, or if there is another way to get at the data.  Or am I missing something all together?

**Comment by JJKraw** [comment]
Can you infer that from the minutes portion of the timestamp field?

**Reply by maplegilbs** [reply]
Unfortunately no - well not unless I know the previous time the accumulator was reset for each station.

So for example it appears at the Burlington station the hourly precipitation is reset to 0 at 54 minutes past the hour (and I don't know this with certainty, or whether or not they change this with any frequency), but other stations that number is different, for example I have seen other stations where it appears that number is reset at 51 minutes past the hour.

At this point the best solution I can come up with without further information or ideas is to make assumptions and just take the observation closest to the on the hour but not past (i.e. the last reading up to but not after 11:59 will give me the hourly precip between 11:00 and 12:00).  But I'd really rather not do that.

**Comment by JJKraw** [comment]
METAR stations (even predating the API) tend to report at the same interval within 10 minutes of the top of the hour so that a "top of the hour" report is available. Although there is no guarantee, from observation I've seen them be pretty consistent. But note that precipitation measurements aren't a part of the "rawMessage" METAR (although I assume taken at the same times).

What led you to believe that the value given isn't just the previous 60 minutes (or 3 / 6 hours) from the timestamp value?

**Comment by maplegilbs** [comment]
Thanks for the info, confirmation from a NOAA employee that at least at the BTV station rainfall data is reported at 54 past the hour and the accumulator is reset at that point.  I will work off that assumption for the other stations (using their minute info from the timestamp).

I believe the value given when reported on those hourly intervals is for the prior 60 minutes, i.e precipitationLastHour value from 12:54 is for the 60 minutes prior to 12:54.  However it appears to me when the value is reported in a shorter timeframe during a rain event, it is only for the period since the rain accumulation total was reset.  So in the case of the BTV station whatever the duration between 54 minutes past the prior hour and the current report timestamp.  Example here:

`"timestamp": "2021-09-08T23:09:00+00:00",`
`"rawMessage": "KBTV 082309Z 17008KT 6SM -TSRA BR SCT049CB BKN100 BKN200 18/17 A2973 RMK AO2 LTG DSNT SE AND S OCNL LTGIC S TS S MOV NE P0004 T01830172 $",`
` "precipitationLastHour": {
                    "value": 0.001,
                    "unitCode": "unit:m",
                    "qualityControl": "qc:C"
                }`

`"timestamp": "2021-09-08T22:54:00+00:00",`
 `"rawMessage": "KBTV 082254Z 17010KT 2 1/2SM TSRA BR FEW023CB OVC047 18/17 A2973 RMK AO2 LTG DSNT NE-S SLP065 OCNL LGTIC S-W TS S-W MOV NE P0086 T01830172",`
`"precipitationLastHour": {
                    "value": 0.0218,
                    "unitCode": "unit:m",
                    "qualityControl": "qc:C"
                },`

`"timestamp": "2021-09-08T22:44:00+00:00",`
 `"rawMessage": "KBTV 082244Z 16009KT 2 1/2SM TSRA BR FEW023 BKN047CB OVC060 18/17 A2972 RMK AO2 LTG DSNT NE AND S AND SW OCNL LGTIC W NE-S-SE TS W NE-E-S-SE MOV E P0077 T01780167",`
`"precipitationLastHour": {
                    "value": 0.019599999999999999,
                    "unitCode": "unit:m",
                    "qualityControl": "qc:C"
                },`

The precipitationLastHour value given at 23:09 is .001.  However the precipitationLastHour difference between 22:44 and 22:54 is .0218-.0196 = .0022, so at least .0022m of rain must have fallen between 22:44 and 22:54, which if the precipitationLastHour value from 23:09 was actually reporting the prior 60 minutes, would have to include that interval between 22:44 and 22:54.  Because the value given of .001 is < the .0022 we know that period is not included in the reported total and thus the value must only go back to the last 'reset' time which I believe to be (and was confirmed by the NOAA employee at BTV) 22:54.

I'm sure there is a reason for doing things this way , but using the key of precipitationLastHour makes this value a bit confusing/misleading for me, but now that I've updated my understanding on it, I can at least update the script to be accurate.

Does that sound like solid reasoning or perhaps a hole in the logic?
Thanks for your insight on this

**Reply by JJKraw** [reply]
Sounds like it could be right. Would be nice to know if all stations are doing the same thing.
